service/src/java/org/apache/hive/service/auth/ldap/LdapSearch.java

```diff
@@ -180,6 +182,8 @@ public List<String> executeCustomQuery(String query) throws NamingException {
    */
   @Override
   public List<String> executeUserAndGroupFilterQuery(String user, String userDn, String groupSearchFilter, String groupBaseDn) throws NamingException {
+    userDn = LdapEncoder.filterEncode(userDn);
+    userDn = Matcher.quoteReplacement(userDn);
```

Comment:
why do we need this?

Reply:
I think refactoring would not add anything here, just make it unreadable. If anybody in the future checks this code it will be clear in 2 seconds what string transformation steps happening here. With this Function introduction it is hard to understand what is actually going on.

Reply:
I disagree here, the function shows the transformation applied to the input arg, whereas changing the input arg itself is a code smell.